app/experimenter/projects/forms.py

```diff
@@ -0,0 +1,37 @@
+from django import forms
+from django.utils.text import slugify
+
+from experimenter.projects.models import Project
+
+
+class ProjectForm(forms.ModelForm):
+    name = forms.CharField(
+        widget=forms.TextInput(attrs={'class': 'form-control'}))
+    slug = forms.CharField(required=False)
+
+    class Meta:
+        model = Project
+        fields = ('name', 'slug', 'image')
+
+    def clean_name(self):
+        name = self.cleaned_data['name']
+        slug = slugify(name)
+
+        if (
+            self.instance.pk is None and
+            self.Meta.model.objects.filter(slug=slug).exists()
+        ):
+            raise forms.ValidationError(
+                'A project with this name already exists.')
+
+        return name
+
+    def clean(self):
+        cleaned_data = super().clean()
+
+        name = cleaned_data.get('name', None)
+
+        if name is not None:
```

Comment:
Do you want to throw an error in `clean_name` so this can't happen?

Reply:
The field is required, so the form will throw that error automatically, so we can remove this check as long as the code still executes for a possible None value.